### PR TITLE
feat(`create-require-from-path`): introduce

### LINKED ISSUE #95: feat: `createRequireFromPath` to `createRequire`
## Description

This codemod should migrate `createRequireFromPath` to `createRequire`. It's useful to migrate `v12` to `v14` or `v16`.

It's should handle both `require()` function and `import` statement.
It's should support both `module` `node:module` specifier.

## Example

**Before:**
```js
const { createRequireFromPath } = require('module');

// Using createRequireFromPath
const requireFromPath = createRequireFromPath('/path/to/module');
const myModule = requireFromPath('./myModule.cjs');
```

**After:**
```js
 const { createRequire } = require('module');

// Using createRequire with a specific path
const require = createRequire('/path/to/module');
const myModule = require('./myModule.cjs');
```

## REFS

https://nodejs.org/en/blog/release/v16.0.0
https://github.com/nodejs/node/pull/37201
https://nodejs.org/docs/v22.17.1/api/deprecations.html#dep0130-modulecreaterequirefrompath

### PR DESCRIPTION
**Pull request by AugustinMauroy**
## Related issue

> [!IMPORTANT]
> It's use `@next` codemod CLI

Close #95

### CODE REVIEW

File: recipes/create-require-from-path/src/workflow.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    const varMatch = usage.getMatch("VAR");
+    const argMatch = usage.getMatch("ARG");
+
+    if (varMatch && argMatch) {
+      const oldVarName = varMatch.text();
+      const arg = argMatch.text();
+
+      // Get the original text to preserve semicolons
+      const originalText = usage.text();
+      const hasSemicolon = originalText.endsWith(';');
```
[the remark this answers is not in the document]
Comment (AugustinMauroy) [reply recipes/create-require-from-path/src/workflow.ts:94]: oh yeah right. I was kind of utopics

File: recipes/create-require-from-path/src/workflow.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        }
+      ]
+    }
+  });
+
+  for (const importNode of importNodes) {
+    const text = importNode.text();
+
+    // Check if this import/require includes createRequireFromPath
+    if (text.includes("createRequireFromPath")) {
```
[the remark this answers is not in the document]
Comment (AugustinMauroy) [reply recipes/create-require-from-path/src/workflow.ts:67]: yeah I need to use ast instead of text manipulation

File: recipes/create-require-from-path/tests/input/edge-case.js
[the remark this answers is not in the document]
Comment (AugustinMauroy) [reply recipes/create-require-from-path/tests/input/edge-case.js:1]: Like ?
```js
// createRequire ...
```

[the remark this answers is not in the document]
Comment (AugustinMauroy) [reply recipes/create-require-from-path/tests/input/edge-case.js:1]: I can add it be it's impossible since we use AST to get it

[the remark this answers is not in the document]
Comment (AugustinMauroy) [reply recipes/create-require-from-path/tests/input/edge-case.js:1]: added the test case. It's don't touch to comment because it's a different node
